### PR TITLE
work towards criminalrecords_cartride

### PR DESCRIPTION
**Pull request by Brandon-Huu**
# **HELP ME**

## About the PR

Todo:
- [ ] Adds a PDA cartridge that lists all of a stations criminal records ( excluding history )
- [ ] Spawn all security with criminal record cartridges in their pda

## Why / Balance

People fucking suck at responding to radio when asked "Who's wanted?"
Having to walk all the way back to sec to check records is extremely exhausting especially when the only reason you have to do it is because people refuse to communicate.

## Technical details

What the fuck does this code even do

## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by beck-thompson** [comment]
I think it would also be nice if lawyers also had this!

File: Content.Server/CartridgeLoader/Cartridges/CriminalRecordsCartrideSystem.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using Content.Server.StationRecords.Systems;
+using Content.Shared.CriminalRecords;
+using Content.Shared.CriminalRecords;
+using Content.Shared.Security;
+
+namespace Content.Server.CartridgeLoader.Cartridges;
+
+public sealed class CriminalRecordsCartridgeSystem : EntitySystem
+{
+  [Dependency] private readonly StationRecordsSystem _StationRecords = default!;
```
Comment (TheShuEd) [inline Content.Server/CartridgeLoader/Cartridges/CriminalRecordsCartrideSystem.cs:13]:
```suggestion
  [Dependency] private readonly StationRecordsSystem _stationRecords = default!;
```
```suggestion
  [Dependency] private readyonly StationSystem _stationSystem = default!;
```

File: Content.Server/CartridgeLoader/Cartridges/CriminalRecordsCartrideSystem.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  public override void Initialize()
+  {
+    base.Initialize();
+    SubscribeLocalEvent<CriminalRecordsCartridgeComponent, CartridgeMessageEvent>(OnUiMessage) 
+    SubscribeLocalEvent<CriminalRecordsCartridgeComponent, CartridgeUiReadyEvent>(OnUiReady); 
+    SubscribeLocalEvent<ProgramInstallationAttempt>(OnInstallationAttempt);
+  }
+
+  private void OnUiMessage(EntityUid uid, CriminalRecordsCartridgeComponent comp, CartridgeMessageEvent args)
```
Comment (TheShuEd) [inline Content.Server/CartridgeLoader/Cartridges/CriminalRecordsCartrideSystem.cs:27]:
```suggestion
  private void OnUiMessage(Entity<CriminalRecordsCartridgeComponent> cartridge, ref CartridgeMessageEvent args)
```
use new Entity<> template

File: Content.Server/CartridgeLoader/Cartridges/CriminalRecordsCartrideSystem.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  {
+    UpdateUiState(uid, GetEntity(args.LoaderUid), comp);
+  }
+
+  private void OnUiReady(EntityUid, CriminalRecordsCartridgeComponent comp, CartridgeUiReadyEvent args)
+  {
+    UpdateUiState(uid, args.Loader, comp);
+  }
+  
+  private void UpdateUiState(EntityUid uid, EntityUid loaderUid, CriminalRecordsCartridgeComponent? comp)
```
Comment (TheShuEd) [inline Content.Server/CartridgeLoader/Cartridges/CriminalRecordsCartrideSystem.cs:37]: Entity<> template

**Review by TheShuEd — COMMENTED** [review]
some unrelated syntactic remarks to your problem
:trollface:

**Comment by Hanzdegloker** [comment]
This plus criminal records being able to be sorted by status (wanted, in custody, etc) would be the ultimate Security QoL

**Comment by Brandon-Huu** [comment]
Removed all of the errors. The code actually builds now

File: Content.Server/CartridgeLoader/Cartridges/CriminalRecordsCartrideSystem.cs (hunk 2 above)
Comment (Brandon-Huu) [reply Content.Server/CartridgeLoader/Cartridges/CriminalRecordsCartrideSystem.cs:31]: I don't know how to do this properly. could you show me everything I would need to change :>

**Comment by Brandon-Huu** [comment]
Yeah, I'm probably going to close this and give up

**Comment by Brandon-Huu** [comment]
Might try again when i figure out how to do UI without needing 8 different files

**Pull request closed without merging**